14/umbraco-cms/reference/security/external-login-providers.md

```diff
@@ -512,14 +512,96 @@ export default MyLitView;
 
 {% endtabs %}
 
+### Generic backoffice login provider composer
+A composer and `genericAuthenticationOptions` configuration class to setup the authentication options for the generic authentication provider using dependency injection.
+
+{% code title="GenericBackOfficeExternalLoginComposer.cs" lineNumbers="true" %}
+```csharp
+// this example uses a non existing generic OathProvider NuGet package
+// but any package that uses the Microsoft.AspNetCore.Authentication.OAuth.OathOptions will work
+using AspNet.Security.OAuth.Generic;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Api.Management.Security;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Web.Common.Helpers;
+
+namespace MyUmbracoProject.CustomAuthentication;
+
+public class GenericBackOfficeExternalLoginComposer : IComposer
+{
+    public void Compose(IUmbracoBuilder builder)
+    {
+        // register the generic loginProvider options (auto linking, manual linking, ...)
+        builder.Services.ConfigureOptions<GenericBackOfficeExternalLoginProviderOptions>();
+        // register the generic authtication options (secret, callback, errorhandling, ...)
+        builder.Services.ConfigureOptions<ConfigureGenericAuthenticationOptions>();
+
+        builder.AddBackOfficeExternalLogins(logins =>
+        {
+            logins.AddBackOfficeLogin(
+                backOfficeAuthenticationBuilder =>
+                {
+                    // this Add... method will be part of the OathProvider nuget package you install
+                    backOfficeAuthenticationBuilder.AddGenericProvider(
+                        BackOfficeAuthenticationBuilder.SchemeForBackOffice(GenericOfficeExternalLoginProviderOptions
+                            .SchemeName)!,
+                        options =>
+                        {
+                            // need to give an empty action here for the options pattern configuration to work 🤷
+                            // if you do not wish to use the umbraco default error handling and hardcode all your values instead of injecting them,
+                            // you can set the configuration right here instead.
+                        });
+                });
+        });
+    }
+}
+
+// the ...AuthenticationOptions method will be part of the OathProvider nuget package you install
+// check the Add... method invoked on the backOfficeAuthenticationBuilder to figure out the correct type
+public class ConfigureGenericAuthenticationOptions : IConfigureNamedOptions<GenericAuthenticationOptions>
```

Comment:
am getting the following error for GenericAuthenticationOptions, is the same on your side?

![image](https://github.com/umbraco/UmbracoDocs/assets/83591955/8bf845dd-148b-4912-9cd3-ece7e2848fa8)